src/core/request/engines/composition/test/engine.spec.ts

```diff
@@ -0,0 +1,422 @@
+import Async from 'core/async';
+import request, { globalOpts, RequestError, RequestResponseObject } from 'core/request';
+import { compositionEngine } from 'core/request/engines/composition';
+import { createServer } from 'core/request/engines/composition/test/server';
+
+// eslint-disable-next-line max-lines-per-function
+describe('core/request/engines/composition as request engine', () => {
+	let server: Awaited<ReturnType<typeof createServer>>;
+
+	let
+		api;
+
+	beforeAll(async () => {
+		api = globalOpts.api;
+		globalOpts.api = undefined;
+		server = await createServer(4444);
+	});
+
+	beforeEach(() => {
+		server.clearHandles();
+		server.handles.json1.response(200, {test: 1});
+	});
+
+	afterAll(() => {
+		globalOpts.api = api;
+		server.destroy();
+	});
+
+	it('engine destructor call should lead to triggering the destructors of all the providers created by the engine', async () => {
```

Comment:
Слишком избыточное описание, можно короче:
```suggestion
	it('engine destructor should trigger the destructors of all providers it created', async () => {
```